build.sh

```diff
@@ -1,65 +1,110 @@
 #!/bin/bash
-
 set -e
 
-podman=podman
-if ! which $podman; then
-  podman=docker
+podman=`which podman`
+if [ -z $podman ]; then
+  podman=`which docker`
 fi
 
-if ! which $podman; then
+if [ -z $podman ]; then
   echo "Either podman or docker need to be in PATH for this script to work."
   exit 1
 fi
 
-if [ -z "$1" ]; then
-  echo "usage: $0 <godot branch> <mono version string> [<mono branch> <mono commit hash>]"
+if [ -z "$1" -o -z "$2" ]; then
+  echo "Usage: $0 <godot branch> <mono version> [<mono branch> <mono commit hash>]"
   echo
   echo "Examples: $0 3.1 mono-5.18.1.3"
-  echo "          $0 master mono-6.6.0.160 2019-08 bef1e6335812d32f8eab648c0228fc624b9f8357"
+  echo "	$0 master mono-6.6.0.160 2019-08 bef1e6335812d32f8eab648c0228fc624b9f8357"
+  echo
+  echo "godot branch:"
+  echo "mono version:"
+  echo "	These are combined to form the docker image tag, e.g. 'master-mono-6.6.0.160'."
+  echo "	Git will then clone the branch/tag that matches the mono version."
+  echo 
+  echo "mono branch:"
+  echo "	If specified, git will clone this mono branch/tag instead. Requires specifying a commit."
   echo
+  echo "mono commit:"
+  echo "	If specified, git will check out this commit after cloning."
+  echo 
   exit 1
 fi
 
 godot_branch=$1
 mono_version=$2
 img_version=$godot_branch-$mono_version
+files_root=$(pwd)/files
 mono_commit=
+mono_commit_str=
+
+# If optional Mono git branch and commit hash were passed, use them.
 if [ ! -z "$3" -a ! -z "$4" ]; then
-  # Optional Mono git branch and commit hash were passed,
-  # use that for the git clones.
   mono_version=$3
   mono_commit=$4
+  mono_commit_str="-${mono_commit:0:6}"
+fi
+
+# If mono branch does not start with mono-, prepend it to the folder name.
+if [ ${mono_version:0:5} != "mono-" ]; then
+  mono_root="${files_root}/mono-${mono_version}${mono_commit_str}"
+else
+  mono_root="${files_root}/${mono_version}${mono_commit_str}"
+fi
+
+# Confirm settings
+echo "Docker image tag: ${img_version}"
+echo "Mono branch: ${mono_version}"
+if [ ! -z "$mono_commit" ]; then
+  echo "Mono commit: ${mono_commit}"
 fi
-echo "Building images with version: ${img_version}"
-echo "Mono version used: ${mono_version} ${mono_commit}"
+if [ -e ${mono_root} ]; then
+  mono_exists="(exists)"
+fi
+echo "Mono source folder: ${mono_root} ${mono_exists}"
 echo
 while true; do
-    read -p "Is this correct? [y/n] " yn
-    case $yn in
-        [Yy]* ) break;;
-        [Nn]* ) exit 1;;
-        * ) echo "Please answer yes or no.";;
-    esac
+  read -p "Is this correct? [y/n] " yn
+  case $yn in
+    [Yy]* ) break;;
+    [Nn]* ) exit 1;;
+    * ) echo "Please answer yes or no.";;
+  esac
```

Comment:
Now `./build.sh master mono-6.6.0.160 2019-08 bef1e6335812d32f8eab648c0228fc624b9f8357` will make (and check for) a folder called `files/mono-2019-08-bef1e6`. Also it will say (exists) if found, so you know it won't be downloaded again.